src/wallet/rpcwallet.cpp

```diff
@@ -746,13 +748,26 @@ static UniValue getreceivedbylabel(const JSONRPCRequest& request)
         {
             CTxDestination address;
             if (ExtractDestination(txout.scriptPubKey, address) && IsMine(*pwallet, address) && setAddress.count(address)) {
+                ColorIdentifier colorId;
+                if(address.which() == 3)
+                    colorId = boost::get<CColorKeyID>(address).color;
+                else if(address.which() == 4)
+                    colorId = boost::get<CColorScriptID>(address).color;
+
                 if (wtx.GetDepthInMainChain() >= nMinDepth)
-                    nAmount += txout.nValue;
+                    lable_balance[colorId] += txout.nValue;
             }
         }
     }
 
-    return ValueFromAmount(nAmount);
+    UniValue balances(UniValue::VOBJ);
+    for (auto const& wb : lable_balance) {
+        if (wb.first.type == TokenTypes::NONE)
+            balances.pushKV(CURRENCY_UNIT.c_str(), ValueFromAmount(wb.second));
+        else
+            balances.pushKV(HexStr(wb.first.toVector()).c_str(), wb.second);
```

Comment:
You can use `wb.first.toHexString` instead of `wb.first.toVector()` and `CURRENCY_UNIT`.